### PR TITLE
testing for `geometry.angle_between_two_lines()`

### LINKED ISSUE #23: public objects in `geometry.py`?
Should [`geometry._is_within()`](https://github.com/uscuni/sgeop/blob/4b6dc2361653a9720666bcd37cd2a68d390f9230/sgeop/geometry.py#L18) and [`geometry. voronoi_skeleton()`](https://github.com/uscuni/sgeop/blob/4b6dc2361653a9720666bcd37cd2a68d390f9230/sgeop/geometry.py#L15) really be listed a public functions? I think probably not since it seems that are only used with `artifacts.py`.

### PR DESCRIPTION
**Pull request by jGaboardi**
This MR:
* starts fine-grained testing in `sgeop`, which will (hopefully) make #7 easier done the line.
* resolves #23

### CODE REVIEW

File: sgeop/geometry.py [hunk 1]
```diff
@@ -49,8 +42,17 @@ def _is_within(
...
+def angle_between_two_lines(
+    line1: shapely.LineString, line2: shapely.LineString
+) -> float:
+    """Return the angle between 2 two lines (assuming they share a vertex).
+    Based on ``momepy.coins`` but adapted to shapely lines.
+    """
+
+    lines_distinct = line1 != line2
+    if not lines_distinct:
+        raise ValueError("Input lines are identical - must be distinct.")
```
Comment (martinfleis) [inline sgeop/geometry.py:54]: Can we avoid adding any errors? I spent quite some time ensuring a single erroneous geometry does not bomb the whole algo on an error which indeed causes an issue in the simplification but only minor.  Can we warn and return something?

File: sgeop/geometry.py [hunk 2]
```diff
@@ -59,6 +61,10 @@ def angle_between_two_lines(line1, line2):
     # (lines touch in this point)
     points = collections.Counter([a, b, c, d])
 
+    lines_share_vertex = max(points.values()) > 1
+    if not lines_share_vertex:
+        raise ValueError("Input lines do not share a vertex.")
```
Comment (martinfleis) [inline sgeop/geometry.py:66]: As above.

File: sgeop/geometry.py (hunk 1 above)
Comment (jGaboardi) [reply sgeop/geometry.py:54]: That's probably reasonable. As for a returned value, which of the following would we think are the most realistic:
* `0.0`
* `numpy.inf`
* `numpy.nan`
* something else?

File: sgeop/geometry.py (hunk 2 above)
Comment (jGaboardi) [reply sgeop/geometry.py:66]: As above.

File: sgeop/geometry.py (hunk 1 above)
Comment (martinfleis) [reply sgeop/geometry.py:54]: I'd go with 0. It is a real value that may not break stuff down the line and the angle is 0 anyway.

File: sgeop/geometry.py (hunk 2 above)
Comment (martinfleis) [reply sgeop/geometry.py:66]: Here the returned value is trickier. But let's do 0 for simplicity as well?

Comment (jGaboardi) [reply sgeop/geometry.py:66]: > Here the returned value is trickier.

Yeah... 

> But let's do 0 for simplicity as well?

Sounds like a plan. At least with a healthy warning we can retain our sanity while debugging later if it becomes a problem.

**Comment by jGaboardi** [comment]
[`geometry.angle_between_two_lines()`](https://github.com/uscuni/sgeop/blob/4b6dc2361653a9720666bcd37cd2a68d390f9230/sgeop/artifacts.py#L617) is deep in the code base. Leaving the warning at `stacklevel=2` for now, but probably should update later.

**Comment by martinfleis** [comment]
It may actually pop at different stacklevels do 2 is good.

**Comment by jGaboardi** [comment]
> It may actually pop at different stacklevels do 2 is good.

So are we good to merge or do you still need to review?

**Review by martinfleis — APPROVED** [review]

**Pull request merged**